spec/consensus/light-client.md

```diff
@@ -257,28 +258,60 @@ Towards Lite Client Completeness:
 
 
 ```go
-func Bisection(h1,h2,trustlevel) bool{
-  if CheckSupport(h1,h2,trustlevel) {
-    return true
-  }
-  if h2.Header.height == h1.Header.height + 1 {
-    // we have adjacent headers that are not matching (failed
-    // the CheckSupport)
-    // we could submit evidence here
-    return false
-  }
-  pivot := (h1.Header.height + h2.Header.height) / 2
-  hp := Commit(pivot)
-    // ask a full node for header of height pivot
-  Store(hp)  
-    // store header hp locally
-  if Bisection(h1,hp,trustlevel) {
-    // only check right branch if hp is trusted
-    // (otherwise a lot of unnecessary computation may be done)
-    return Bisection(hp,h2,trustlevel)
+func verify(h) {
+  if hasExpired(h) return false
+
+  vp_all := totalVotingPower(h.Header.V) // total sum of voting power of validators in h
+
+  if votingpower_in(signers(h.Commit),h.Header.V) > 2/3 * vp_all {
+        return hasExpired(h)
```

Comment:
why call hasExpired(h) again if we've already done so above?

Reply:
The idea of calling hasExpired before returning is giving caller duration of lease, i.e., for how long header will be consider valid.

Reply:
Sorry, I do not understand. `hasExpired` returns a boolean (true or false), so for caller it won't make a different how many times we execute `hasExpired` function.

Reply:
My mistake, it should return a pair (boolean, int) for example, where latter is lease duration. Will make a new version to address all comments.